src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -81,81 +81,71 @@ protected virtual Task HandleReceiveExceptionAsync(Exception receiveException)
             return Task.CompletedTask;
         }
 
+        // TODO: move to extension?
+        /// <summary>
+        ///     Determines the encoding used for a given message
+        /// </summary>
+        /// <remarks>If no encoding was specified, UTF-8 will be used by default</remarks>
+        /// <param name="messageContext">Context concerning the message</param>
+        /// <returns>Encoding that was used for the message body</returns>
+        protected virtual Encoding DetermineMessageEncoding(MessageContext messageContext)
```

Comment:
Don't we already have this?

AFAIK I already wrote this

Reply:
Yes, was only removed in the destination PR-branch. Now reverted. I added a TODO though to maybe move this out of the message pump.